### PR TITLE
GCI-246: Replaced the Long constructor with valueOf

### PR DESCRIPTION
**Pull request by shakeelsamsu**
## Description of what I changed

Use Long.valueOf(long) instead of the inefficient new Long(long) constructor

## Issue I worked on

see https://issues.openmrs.org/browse/GCI-246

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateLocationDAO.java [hunk 1]
```diff
@@ -345,7 +345,7 @@ public LocationAttributeType getLocationAttributeTypeByName(String name) {
 		    Projections.rowCount()).add(Restrictions.eqProperty("alias.locationId", "outer.locationId"));
 		
 		return sessionFactory.getCurrentSession().createCriteria(Location.class, "outer").add(
-		    Restrictions.eq("retired", false)).add(Subqueries.eq((long) tags.size(), numberOfMatchingTags)).list();
+		    Restrictions.eq("retired", false)).add(Subqueries.eq(Long.valueOf(tags.size()), numberOfMatchingTags)).list();
```
Comment (samuelmale) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernateLocationDAO.java]: Looks good but could you edit your commit message to more intuitive wording of what you really did not just what you were supposed to do. Starting with something like "GCI-246 : Replaced.."

Comment (shakeelsamsu) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernateLocationDAO.java]: Will do.

**Comment by ykarim** [comment]
Code looks all good but please follow the [OpenMRS Pull Request Tips](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) (take a look at the branch name == ticket name part). The pull request branch should be GCI-246 against the master branch.

**Comment by shakeelsamsu** [comment]
Thanks, I'll keep that in mind - but how do I change that now that I've made the pull request?

**Comment by samuelmale** [comment]
Simply use.
`$ git commit --amend -m "new commit message"`

Then since you have already push the previous commit just do
`$ git push --force example-branch `

You could good about editing a commit message with git.

**Comment by shakeelsamsu** [comment]
> Simply use.
> `$ git commit --amend -m "new commit message"`
> 
> Then since you have already push the previous commit just do
> `$ git push --force example-branch `
> 
> You could google about editing a commit message with git.

I have already changed the commit message. I was wondering on **how I could change the branch I compared from _master_ to _GCI-246_** without making a new pull request.
I'm still new to git, so please bear with me as I try to learn :)

**Comment by shakeelsamsu** [comment]
I think I'll just make a new pull request from the correct branch. Sorry for the misconception.

**Pull request closed without merging**